File: php/decorator/Soy.php
<?
include_once("CondimentDecorator.php");

class Soy extends CondimentDecorator {
    private $beverage;

    public function __construct(Beverage $beverage) {
        $this->beverage = $beverage;
    }

    public function getDescription() {
        return $this->beverage->getDescription() . ", Soy";
    }

    public function cost() {
        return .15 + $this->beverage->cost();
    }
}
?>

File: python/2.7/decorator/Soy.py
from CondimentDecorator import *

class Soy(CondimentDecorator):
    def __init__(self, beverage):
        self.beverage = beverage

    def getDescription(self):
        return self.beverage.getDescription() + ", Soy"

    def cost(self):
        return .15 + self.beverage.cost()